nixos/modules/services/hardware/thermald.nix

```diff
@@ -50,8 +50,8 @@ in
             ${optionalString cfg.debug "--loglevel=debug"} \
             ${optionalString cfg.ignoreCpuidCheck "--ignore-cpuid-check"} \
             ${optionalString (cfg.configFile != null) "--config-file ${cfg.configFile}"} \
-            --dbus-enable \
-            --adaptive
+            ${optionalString (cfg.configFile == null) "--adaptative"} \
```

Comment:
I think using `if` is better since it indicates that only one of `--adaptative` and `--config-file ${cfg.configFile}` will be used:

```nix
${if cfg.configFile != null then "--config-file ${cfg.configFile}" else "--adaptative"}
```

What about also mentioning this in the `description` of `configFile` option?

Reply:
Just find that https://github.com/NixOS/nixpkgs/pull/310819 has been merged to fix the same issue.

If you are willing to implement the improvements I suggested above, then this PR can still be merged.